File: 2562. Find the Array Concatenation Value/Solution.py
from typing import List


class Solution:
    def findTheArrayConcVal(self, nums: List[int]) -> int:
        res, n = 0, len(nums)
        i, j = 0, n - 1
        while i <= j:
            if i == j:
                res += nums[i]
            else:
                res += int(str(nums[i]) + str(nums[j]))
            i += 1
            j -= 1
        return res

File: 2562. Find the Array Concatenation Value/Solution.cpp
#include <vector>
using namespace std;

class Solution {
public:
    long long findTheArrayConcVal(vector<int>& nums)
    {
        long long res = 0;
        int n = nums.size();
        for (int i = 0, j = n - 1; i <= j; i++, j--) {
            if (i == j)
                res += nums[i];
            else {
                for (int x = nums[j]; x; x /= 10)
                    nums[i] *= 10;
                res += nums[i] + nums[j];
            }
        }
        return res;
    }
};
